### PR TITLE
Workflow cleanup

### PR DESCRIPTION
**Pull request by soumeh01**
The separate steps would give detailed results even if the step before gets failed

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -100,12 +104,17 @@ jobs:
             qemu-user-binfmt
 
       - name: Unit testing (for Linux-Arm64)
-        if: ${{ startsWith(runner.os, 'Linux') }}
+        if: ${{ startsWith(runner.os, 'Linux') && (success() || failure()) }}
```
Comment (JonatanAntoni) [inline .github/workflows/test.yml:107]: According to https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions the alternative to `always()` seems to be `!cancelled()` (which should be equivalent to `success() || failure()` but shorted)

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -100,12 +104,17 @@ jobs:
...
         run: |
           GOOS=linux GOARCH=arm64 go test -v ./... > build/cbuildtests-${{ matrix.target }}-arm64.txt
+
+      - name: Generate JUnit test report (for Linux-Arm64)
+        if: ${{ startsWith(runner.os, 'Linux') && (success() || failure()) }}
+        run: |
           go-junit-report -set-exit-code -in build/cbuildtests-${{ matrix.target }}-arm64.txt -iocopy -out build/cbuild-testreport-${{ matrix.target }}-arm64.xml
 
       - name: Archive unit test results
+        if: always()
```
Comment (JonatanAntoni) [inline .github/workflows/test.yml:117]: If we do not run the report generation, we probably don't need to run the upload as well.

**Review by JonatanAntoni — DISMISSED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (soumeh01) [reply .github/workflows/test.yml]: Yes, you are right.
Corrected this [Addressed review comments](https://github.com/Open-CMSIS-Pack/cbuild/pull/78/commits/d8ebc2fd7bf590b252b1e25ed3b79709d5ca63d9)

File: .github/workflows/test.yml (hunk 2 above)
Comment (soumeh01) [reply .github/workflows/test.yml]: [Addressed review comments](https://github.com/Open-CMSIS-Pack/cbuild/pull/78/commits/d8ebc2fd7bf590b252b1e25ed3b79709d5ca63d9)

**Review by JonatanAntoni — APPROVED** [review]

**Pull request merged**